### ISSUE TITLE
Add end-to-end integration smoke test (1-epoch training run)

### ISSUE THREAD

**Issue by luipin** [luipin/set-ml-module#22]
### Summary
The test suite consists entirely of unit tests. There is no integration test that wires `SetCardDataModule` + `MultiHeadResNet` + `pl.Trainer` together to confirm the full pipeline runs end-to-end without errors and that the loss decreases in the expected direction.

Integration bugs — such as a shape mismatch between the DataModule output and the model input, or a misconfigured scheduler — would not be caught by the existing unit tests.

### Proposed test
Add `tests/test_integration.py` with a single smoke test:

```python
def test_one_epoch_training_runs_without_error(tmp_path):
    """Confirm the full DataModule → Model → Trainer pipeline runs for 1 epoch."""
    # Write a minimal set of valid card images
    _write_minimal_dataset(tmp_path, n_per_class=4)

    dm = SetCardDataModule(data_dir=str(tmp_path), batch_size=4, num_workers=0)

    with patch_resnet_weights():
        model = MultiHeadResNet(freeze_epochs=0)

    trainer = pl.Trainer(max_epochs=1, enable_progress_bar=False, logger=False)
    trainer.fit(model, dm)

    # Loss should be finite (not NaN/Inf)
    assert trainer.callback_metrics["train_loss_epoch"].isfinite()
```

### Why this matters
A 1-epoch integration test is cheap (seconds) but catches an entire class of bugs — broken batch format, wrong tensor device, misconfigured optimizer, scheduler step count mismatch — that unit tests cannot catch in isolation.

### Files to create
- `tests/test_integration.py`
